src/models/BaseModel.py

```diff
@@ -212,6 +214,7 @@ def calculate_error(self, train_result, test_result):
         """
         train_error = sklearn.metrics.mean_absolute_error(self.y_train, train_result)
         test_error = sklearn.metrics.mean_absolute_error(self.y_test, test_result)
+        self.mae_test_error = test_error
```

Comment:
If you incorporate train error and convert to 2 legend bar chart can you remove self.mae_test_error and instead define train_error and test_error like you defined mae_test_error so that we avoid unnecessary variable duplication ?